src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+            TileEntity tileentity = null;
+            if (block.hasTileEntity(iblockstate))
+            {
+                tileentity = block.createTileEntity(clientWorld, iblockstate);
+                tileentity.readFromNBT(aBlockList.tileentityData);
+            }
+            ModelHolder models = new ModelHolder(blockpos, iblockstate, iBlockExtendedState, tileentity, ibakedmodel);
+            this.getQuads(models, models.quads);
+            this.renderGhost(clientWorld, models, player, partialTicks);
+        }
+
+        for (final Entity anEntityList : entityList)
+        {
+            Minecraft.getMinecraft().getRenderManager().renderEntityStatic(anEntityList, 0.0F, true);
+        }
+    }
+
+    private void renderGhost(final World world, ModelHolder holder, final EntityPlayer player, final float partialTicks)
+    {
+        boolean existingModel = !this.mc.theWorld.isAirBlock(holder.pos);
+
+        IBlockState actualState = holder.actualState;
+        Block block = actualState.getBlock();
+
+        if (actualState.getRenderType() == EnumBlockRenderType.MODEL)
+        {
+            BlockRenderLayer originalLayer = MinecraftForgeClient.getRenderLayer();
+
+            for (BlockRenderLayer layer : BlockRenderLayer.values())
+            {
+                if (block.canRenderInLayer(actualState, layer))
+                {
+                    this.mc.getTextureManager().bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    ForgeHooksClient.setRenderLayer(layer);
+                    this.renderGhostBlock(world, holder, player, layer, existingModel, partialTicks);
+                    holder.rendered = true;
+                }
+            }
+
+            ForgeHooksClient.setRenderLayer(originalLayer);
+        }
+
+        if (holder.te != null && !holder.rendered)
+        {
+            TileEntity te = holder.te;
+            te.setPos(holder.pos);
+            FakeWorld fakeWorld = new FakeWorld(holder.actualState, world.getSaveHandler(), world.getWorldInfo(), world.provider, world.theProfiler, true);
+            te.setWorldObj(fakeWorld);
+            int pass = 0;
+
+            if (te.shouldRenderInPass(pass))
+            {
+                TileEntityRendererDispatcher terd = TileEntityRendererDispatcher.instance;
+                terd.func_190056_a(fakeWorld,
+                        Minecraft.getMinecraft().renderEngine,
+                        Minecraft.getMinecraft().fontRendererObj,
+                        new FakeEntity(fakeWorld),
+                        (RayTraceResult) null,
+                        0.0F);
+                GL11.glPushMatrix();
+                terd.renderEngine = Minecraft.getMinecraft().renderEngine;
+                terd.preDrawBatch();
+                GL11.glColor4f((float) (this.COLOR_R / COLOR_R), (float) (this.COLOR_G / COLOR_G), (float) (this.COLOR_B / COLOR_B), (this.COLOR_A / COLOR_A));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessParenthesesCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)